vital/tests/test_mesh_io.cxx

```diff
@@ -0,0 +1,184 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief test core mesh_io functionality
+
+#include <kwiversys/SystemTools.hxx>
+#include <tests/test_gtest.h>
+#include <tests/test_scene.h>
+#include <arrows/core/mesh_operations.cxx>
+#include <vital/exceptions.h>
+
+#include <vital/io/mesh_io.cxx>
+
+#include <gtest/gtest.h>
+
+kwiver::vital::path_t g_data_dir;
+
+// ----------------------------------------------------------------------------
+int main(int argc, char** argv)
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  GET_ARG(1, g_data_dir);
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+class mesh_io : public ::testing::Test
+{
+  TEST_ARG(data_dir);
+
+  void compare_meshes(const kwiver::vital::mesh_sptr& first,
+                      const kwiver::vital::mesh_sptr& second)
+  {
+    kwiver::vital::mesh_vertex_array<3>& first_vertices = dynamic_cast
+      <kwiver::vital::mesh_vertex_array<3>&>( first->vertices() );
+    kwiver::vital::mesh_vertex_array<3>& second_vertices = dynamic_cast
+      <kwiver::vital::mesh_vertex_array<3>&>( second->vertices() );
+
+    EXPECT_EQ( first_vertices.size(), second_vertices.size() );
+    for(unsigned int i=0; i<first_vertices.size(); i++)
+    {
+      EXPECT_EQ( first_vertices[i].size(), second_vertices[i].size() );
+      for(unsigned int j=0; j<first_vertices[i].size(); j++)
+      {
+        EXPECT_EQ( first_vertices[i][j], second_vertices[i][j] );
+      }
+    }
+
+    kwiver::vital::mesh_face_array first_faces = *std::make_shared
+      <kwiver::vital::mesh_face_array>(first->faces());
+    kwiver::vital::mesh_face_array second_faces = *std::make_shared
+      <kwiver::vital::mesh_face_array>(second->faces());
+
+    EXPECT_EQ( first_faces.size(), second_faces.size() );
+    for(unsigned int i=0; i<first_faces.size(); i++)
+    {
+      EXPECT_EQ( first_faces[i].size(), second_faces[i].size() );
+      for(unsigned int j=0; j<first_faces[i].size(); j++)
+      {
+        EXPECT_EQ( first_faces[i][j], second_faces[i][j] );
+      }
+    }
+  }
+};
+
+// ----------------------------------------------------------------------------
+TEST_F(mesh_io, invalid_output_file)
+{
+  EXPECT_NO_THROW( kwiver::vital::check_output_file( "temp/name" ) );
+  EXPECT_THROW( kwiver::vital::check_output_file( "temp" ),
+    kwiver::vital::file_write_exception );
+  kwiversys::SystemTools::RemoveADirectory( "temp" );
+}
+
+// ----------------------------------------------------------------------------
+TEST_F(mesh_io, invalid_input_file)
+{
+  EXPECT_THROW( kwiver::vital::check_input_file( data_dir + "/nonexistant" ),
+    kwiver::vital::file_not_found_exception );
+  EXPECT_THROW( kwiver::vital::check_input_file( data_dir + "/videos" ),
```

Comment:
Is this second `EXPECT_THROW` testing something different from the first?

Reply:
yes, one tests the exception for passing in a nonexistant file, the other tests the exception for passing in a directory